assets/js/sidebar-v2.js

```diff
@@ -37,6 +37,15 @@ document.addEventListener('click', (e) => {
   }
 });
 
+document.addEventListener('DOMContentLoaded', () => {
+  const currentPage = window.location.href;
+  const match = Array.from(document.querySelectorAll('.sidebar__link')).find(
+    (el) => el.href.includes(currentPage)
+  );
```

Comment:
This is 90% there @maximusmith529 👍 .
I found an edge case, where if the page you're currently on has a toc (table of contents), it won't scroll into view.
This happens because non-toc side bar items are `<a>` with a href, but toc items are `<button>`'s with a dropdown. You can test this out with the[ table variations page](https://frontdoor-test-docs.nginx.com/previews/nginx-hugo-theme/391/test-product/tables/variations/), since that has a toc.

My suggestion is to use the `id` from the element, instead of a href. This will be available regardless of type, and _always_ contains the url path, hence `window.location.href` -> `window.location.pathname`.
I've also changed the class selector to `.sidebar__link--current` as it's used in both `a` and `button` cases.

```suggestion
  const currentPage = window.location.pathname;
  const match = Array.from(document.querySelectorAll('.sidebar__link--current')).find(
    (el) => el?.id?.includes(currentPage)
  );
```